onnxruntime/core/optimizer/gemm_sum_fusion.cc

```diff
@@ -0,0 +1,167 @@
+  new_gemm_node.AddAttribute("alpha", alpha);
+  new_gemm_node.AddAttribute("beta", beta);
+
+  // Move both input edges from original gemm to new gemm.
+  for (auto gemm_input_edge : graph_utils::GraphEdge::GetNodeInputEdges(gemm_node)) {
+    ORT_ENFORCE(gemm_input_edge.src_arg_index < 2);
+    graph.AddEdge(gemm_input_edge.src_node, new_gemm_node.Index(), gemm_input_edge.src_arg_index, gemm_input_edge.dst_arg_index);
+    graph.RemoveEdge(gemm_input_edge.src_node, gemm_input_edge.dst_node, gemm_input_edge.src_arg_index, gemm_input_edge.dst_arg_index);
+  }
+
+  // Move all output edges from sum to new gemm.
+  for (auto sum_output_edge : graph_utils::GraphEdge::GetNodeOutputEdges(sum_node)) {
+    ORT_ENFORCE(sum_output_edge.src_arg_index == 0);
+    graph.AddEdge(new_gemm_node.Index(), sum_output_edge.dst_node, sum_output_edge.src_arg_index, sum_output_edge.dst_arg_index);
+    graph.RemoveEdge(sum_output_edge.src_node, sum_output_edge.dst_node, sum_output_edge.src_arg_index, sum_output_edge.dst_arg_index);
+  }
+
+  // Finally, move the other sum input edge to "C" for the new gemm node.
+  // If The other sum input def is a a graph input, there is no edge to move.
+  bool sum_input_moved = false;
+  for (auto sum_input_edge : graph_utils::GraphEdge::GetNodeInputEdges(sum_node)) {
+    // The C tensor in GemmSumFusion's documentation is the sum output which does not come from
+    // the fused gemm. The following condition will be true when seeing the input of C to sum.
+    if (sum_input_edge.src_node != gemm_node.Index()) {
+      ORT_ENFORCE(!sum_input_moved);
+      graph.AddEdge(sum_input_edge.src_node, new_gemm_node.Index(), sum_input_edge.src_arg_index, 2);
+      graph.RemoveEdge(sum_input_edge.src_node, sum_input_edge.dst_node, sum_input_edge.src_arg_index, sum_input_edge.dst_arg_index);
+      sum_input_moved = true;
+    }
+  }
+  // Old gemm node output is no longer needed. It was previously fed into the
+  // sum node which is now also handled by the new gemm. Remove this output edge
+  // to allow the node to be removed from the graph.
+  graph_utils::RemoveNodeOutputEdges(graph, gemm_node);
+  ORT_ENFORCE(graph.RemoveNode(gemm_node.Index()));
+
+  // All output edges of the sum node should have been removed already, so we can
+  // remove it from the graph.
+  ORT_ENFORCE(sum_node.GetOutputEdgesCount() == 0);
+  ORT_ENFORCE(graph.RemoveNode(sum_node.Index()));
+
+  modified = RewriteRuleEffect::kRemovedCurrentNode;
+
+  return Status::OK();
+}
+
+bool GemmSumFusion::SatisfyCondition(const Graph& graph, const Node& node, const logging::Logger&) const {
+  // Perform a series of checks. If any fail, fusion may not be performed.
+
+  // Original gemm's C must be missing for this fusion pattern to be valid.
+  // Supported for Opset >=11 as earlier opsets have C as a required input
+  if (!graph_utils::IsSupportedOptypeVersionAndDomain(node, "Gemm", {11, 13}) ||
```

Comment:
[nit] Not sure if we should extend 13 to infinite.

Reply:
Is this in reference to the documentation or code?

Reply:
Code.